lib/ex_unit/test/ex_unit/case_test.exs

```diff
@@ -112,4 +114,29 @@ defmodule ExUnit.CaseTest do
       end
     end
   end
+
+  test "warns for using it twice with different options" do
+    assert capture_io(:stderr, fn ->
+             defmodule WarnsUsedTwice do
+               use ExUnit.Case
+               use ExUnit.Case, async: true
+             end
+           end) == ""
+
+    stderr =
+      capture_io(:stderr, fn ->
+        defmodule WarnsUsedTwice do
+          use ExUnit.Case
+          use ExUnit.Case, async: false
+        end
+      end)
+
+    assert stderr =~ """
+           ExUnit.Case was already used on this module ExUnit.CaseTest.WarnsUsedTwice \
+           with the options `[async: true]`.
+
+           Please make sure to set the right options on the first `use ExUnit.Case`, \
+           as they are not overriden by any subsequent calls.
+           """
```

Comment:
There's no need to match on the whole exact message. Let's just match on

```elixir
assert stderr =~ "ExUnit.Case was already used"
```

Reply:
Sure 👍